### PR TITLE
fix: Update to major version of @vercel/edge-config

### LINKED ISSUE #379: Update @vercel/edge-config dependency
**Is your feature request related to a problem? Please describe.**

No problem in particular, but it would be great to have the dependency `@vercel/edge-config` being updated to latest version as it has some breaking changes causing a type issue with `init` argument

**Describe the solution you'd like**

Update `@vercel/edge-config dependency` to v1 🙏🏻

### PR DESCRIPTION
**Pull request by yusinto**
Fixes #379 . I have tested this locally and verified the type errors are fixed and the example application (complete) still works.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,8 @@
     "packages/sdk/react-native",
     "packages/sdk/react-native/example",
     "packages/sdk/vercel",
+    "packages/sdk/vercel/examples/complete",
+    "packages/sdk/vercel/examples/route-handler",
```
Comment (yusinto) [inline package.json:17]: Include examples in the workspace to take advantage of the monorepo setup.

File: packages/sdk/vercel/examples/complete/middleware.ts
Comment (yusinto) [inline packages/sdk/vercel/examples/complete/middleware.ts:1]: No functional changes here just minor refactoring for the sake of my ocd.

File: packages/sdk/vercel/examples/complete/package.json [hunk 2]
```diff
@@ -10,16 +10,16 @@
     "start": "next start"
   },
   "dependencies": {
-    "@launchdarkly/vercel-server-sdk": "1.0.0",
-    "@vercel/edge-config": "0.1.10",
+    "@launchdarkly/vercel-server-sdk": "1.3.0",
+    "@vercel/edge-config": "^1.1.0",
```
Comment (yusinto) [inline packages/sdk/vercel/examples/complete/package.json:14]: These are the main updates.

File: packages/sdk/vercel/examples/route-handler/next.config.js
Comment (yusinto) [inline packages/sdk/vercel/examples/route-handler/next.config.js:1]: `appDir` is now part of @vercel/edge-config and no longer experimental. Hence this file is unnecessary.

File: packages/sdk/vercel/examples/route-handler/package.json [hunk 3]
```diff
@@ -9,9 +9,9 @@
     "lint": "next lint"
   },
   "dependencies": {
-    "@launchdarkly/vercel-server-sdk": "1.0.0",
-    "@vercel/edge-config": "^0.2.1",
-    "next": "^13.4.4",
+    "@launchdarkly/vercel-server-sdk": "1.3.0",
+    "@vercel/edge-config": "^1.1.0",
+    "next": "14.1.2",
```
Comment (yusinto) [inline packages/sdk/vercel/examples/route-handler/package.json:14]: Major updates here too.

File: packages/sdk/vercel/package.json [hunk 4]
```diff
@@ -33,11 +33,11 @@
     "prettier": "prettier --write '**/*.@(js|ts|tsx|json|css)' --ignore-path ../../../.prettierignore",
     "test": "NODE_OPTIONS=\"--experimental-vm-modules --no-warnings\" jest --ci --runInBand",
     "coverage": "yarn test --coverage",
-    "check": "yarn prettier && yarn lint && yarn build && yarn test && yarn doc"
+    "check": "yarn prettier && yarn lint && yarn build && yarn test"
```
Comment (yusinto) [inline packages/sdk/vercel/package.json:36]: `doc` never exists in this project, so this was breaking.

File: packages/sdk/vercel/src/utils/mockEdgeConfigClient.ts [hunk 5]
```diff
@@ -5,6 +5,13 @@ const mockEdgeConfigClient: EdgeConfigClient = {
   getAll: jest.fn(),
   digest: jest.fn(),
   has: jest.fn(),
+  connection: {
+    baseUrl: '',
+    id: '',
+    token: '',
+    version: '',
+    type: 'vercel',
+  },
```
Comment (yusinto) [inline packages/sdk/vercel/src/utils/mockEdgeConfigClient.ts:14]: This is newly added in the major version of edge-config.

File: packages/sdk/vercel/examples/complete/middleware.ts [hunk 6]
```diff
@@ -8,32 +8,35 @@ export const config = {
...
+  { headers, method, nextUrl, url }: NextRequest,
+  _context: NextFetchEvent,
+) {
   // for demo purposes, warn when there is no EDGE_CONFIG or LAUNCHDARKLY_CLIENT_SIDE_ID
   if (
     !process.env.EDGE_CONFIG ||
     !process.env.LD_CLIENT_SIDE_ID ||
     !parseConnectionString(process.env.EDGE_CONFIG)
   ) {
-    return NextResponse.rewrite(new URL('/missing-edge-config', request.url));
```
Comment (yusinto) [inline packages/sdk/vercel/examples/complete/middleware.ts:18]: I'm not sure how this example could have worked? `request` does not exist.

File: packages/sdk/vercel/examples/complete/middleware.ts [hunk 7]
```diff
@@ -43,20 +46,20 @@ export async function middleware(req: NextRequest, context: NextFetchEvent) {
       );
 
       return hotDogFaviconEnabled
```
Comment (yusinto) [inline packages/sdk/vercel/examples/complete/middleware.ts:48]: Same applies to `request` here as [above](https://github.com/launchdarkly/js-core/pull/393/files#r1517039902).

**Comment by kinyoklion** [comment]
If you want to release this, versus bundling it with other changes in the future, then don't use chore. If we want a releasable unit that isn't `fix`, `fix!`, `feat`, or `feat!`, then we could configure one. But it is probably easiest to choose one of those.

**Comment by yusinto** [comment]
This should be a feat, because it breaks the previous version types.

**Comment by kinyoklion** [comment]
> This should be a feat, because it breaks the previous version types.

A customer will need to update their code with this upgrade? If so `feat!:` The exclamation point indicates breaking changes.

**Comment by yusinto** [comment]
@kinyoklion I changed this to fix because I tested this PR and it does not break existing usage. If anything, it fixes the EdgeConfigClient type mismatch in `init` as described in #379 .

**Comment by kinyoklion** [comment]
I know it isn't related to this PR, but is detox e2e unstable? Because of how our releases work we need to be extra careful that CI jobs are stable.

**Comment by yusinto** [comment]
> is detox e2e unstable

It used to be stable but it has declined to be not stable. It's very disappointing but I may have to disable/exclude it completely, or else only run it when RN changes are made. I'll create a separate PR to fix this.

**Comment by yusinto** [comment]
The detox tests are definitely flaky in CI although they are stable locally when run through `act`. It's disappointing. It's successful here now, but I'll make it non-mandatory/remove it in a separate PR.

**Review by kinyoklion — APPROVED** [review]

**Pull request merged**